File: src/main/java/com/sogeti/codingchallenge/day6/model/Group.java
package com.sogeti.codingchallenge.day6.model;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Group {

    private final List<Character> answers;
    private final List<List<Character>> personsInGroupAnswers;

    public Group(List<Character> answers, List<List<Character>> personsInGroupAnswers) {
        this.answers = answers;
        this.personsInGroupAnswers = personsInGroupAnswers;
    }

    public int amountAnyoneAnsweredYes() {
        Set<Character> answeredYes = new HashSet<>(answers);
        return answeredYes.size();
    }

    public Set<Character> getGivenAnswerSet() {
        return new HashSet<>(answers);
    }

    public int amountEveryoneAnsweredYes() {
        int amountAnswersEveryoneAnsweredYes = 0;
        for (Character answer : getGivenAnswerSet()) {
            boolean everyOneAnsweredYes = true;
            for (List<Character> personAnswers : personsInGroupAnswers) {
                if (!personAnswers.contains(answer)) {
                    everyOneAnsweredYes = false;
                    break;
                }
            }
            if (everyOneAnsweredYes) amountAnswersEveryoneAnsweredYes++;
        }
        return amountAnswersEveryoneAnsweredYes;
    }

    @Override
    public String toString() {
        return "Group{" +
                "answers=" + answers +
                ", personsInGroupAnswers=" + personsInGroupAnswers +
                '}';
    }
}

File: src/main/java/hzt/aoc/day06/model/Group.kt
package hzt.aoc.day06.model

class Group(private val answers: List<Char>, private val personsInGroupAnswers: List<List<Char>>) {
    fun amountAnyoneAnsweredYes(): Int {
        val answeredYes: Set<Char> = HashSet(answers)
        return answeredYes.size
    }

    val givenAnswerSet: Set<Char>
        get() = HashSet(answers)

    fun amountEveryoneAnsweredYes(): Int {
        var amountAnswersEveryoneAnsweredYes = 0
        for (answer in givenAnswerSet) {
            var everyOneAnsweredYes = true
            for (personAnswers in personsInGroupAnswers) {
                if (!personAnswers.contains(answer)) {
                    everyOneAnsweredYes = false
                    break
                }
            }
            if (everyOneAnsweredYes) amountAnswersEveryoneAnsweredYes++
        }
        return amountAnswersEveryoneAnsweredYes
    }

    override fun toString(): String {
        return "Group{" +
                "answers=" + answers +
                ", personsInGroupAnswers=" + personsInGroupAnswers +
                '}'
    }
}
